src/axom/inlet/Inlet.hpp

```diff
@@ -119,6 +119,23 @@ class Inlet
    */
   Table& addTable(const std::string& name, const std::string& description = "");
 
+  /*!
+   *****************************************************************************
+   * \brief Add a structure to the input file schema.
+   *
+   * Adds a structure/record to the input file schema. Structures can contain
+   * fields and/or substructures.  By default, it is not required unless marked with
```

Comment:
Does it also support lists? I'm not familiar enough with the usage of "field" here to know for sure, but it might be clear from the broader context of the class. I don't know if a change is needed to the documentation, but I just want to make sure lists are supported.

Reply:
If I'm understanding your question, the intent is that lists of structs would be added via `addGenericArray`, and lists of primitives would be added via `addIntArray`, `addStringArray`, etc.  Maybe `addStructArray`/`addStructDictionary` would be clearer names that match the naming used for this new `addStruct` function?

Reply:
My question was meant to be about the documentation more than anything. It says "Structures can contain fields and/or substructures". Since substructures are called out explicitly, it made me wonder if it does not support lists. Sounds like it does, though.

As far as your suggestion goes, I think the more consistent the names are, the better. Given that we have e.g. `addString()/addStringArray()`, having `addStruct()/addStructArray()` would be nice.

Reply:
I think for this PR I can add `addStructArray` et all as aliases for `addGenericArray`.  Since a follow-up PR will be removing `addTable` from the public interface of `inlet::Table`, I think the removal of `addGeneric*` can be made part of that PR.